cardano_node_tests/tests/test_scripts.py

```diff
@@ -1914,69 +1917,115 @@ def payment_addrs(
 
         return addrs
 
-    @allure.link(helpers.get_vcs_link())
-    @common.PARAM_USE_BUILD_CMD
-    @pytest.mark.parametrize("script_version", ("simple_v1", "simple_v2"))
-    @pytest.mark.dbsync
-    def test_script_reference_utxo(
+    @pytest.fixture
+    def create_reference_script(
```

Comment:
What is the advantage of moving this to fixture?

Reply:
To be sure that the dbsync check is always looking to the reference script used since:

>  a script is added to `script` table only the first time it is seen, so the record can be empty for the current transaction

with a fixture we only create the reference script once and we use it on the different cases, so when we use the function check_tx in this test we will always check the scripts txins.

Reply:
I still don't get why we need to make sure `check_tx` always checks the scripts txins. The `check_tx` needs to be general enough to handle both scenarios:
* scripts added by the transaction we are checking
* scripts that are referenced, but that were already added by previous transactions